utils/s2n_array.c

```diff
@@ -103,6 +116,7 @@ S2N_RESULT s2n_array_insert(struct s2n_array *array, uint32_t index, void **elem
         /* Enlarge the array */
         uint32_t new_capacity = 0;
         GUARD_AS_RESULT(s2n_mul_overflow(current_capacity, 2, &new_capacity));
+        new_capacity = MAX(new_capacity, S2N_INITIAL_ARRAY_SIZE);
```

Comment:
You need to update the check you removed from the harnesses to map this behavior. Actually, you probably only need to update the `s2n_array_pushback` proof harness.